mcpserver/handlers.go

```diff
@@ -631,3 +658,51 @@ func parseJobID(jobID string) (cluster int, proc int, err error) {
 
 	return cluster, proc, nil
 }
+
+// parseJWTUsername extracts the username (sub claim) from a JWT token using the JWT library
+// Returns the username or an error if parsing fails
+func parseJWTUsername(token string) (string, error) {
```

Comment:
Combine `parseJWTUsername` and `parseJWTExpiration` into a single function to avoid parsing the same token multiple times.